modules/edge-identity/src/identity.rs

```diff
@@ -65,47 +65,47 @@ pub enum IdentityStage {
 #[derive(Encode, Decode, PartialEq)]
 pub struct IdentityRecord<AccountId, BlockNumber> {
 	pub account: AccountId,
+	pub identity_type: IdentityType,
 	pub identity: Vec<u8>,
 	pub stage: IdentityStage,
 	pub expiration_time: Option<BlockNumber>,
 	pub proof: Option<Attestation>,
-	pub verifications: Option<Vec<(AccountId, bool)>>,
 	pub metadata: Option<MetadataRecord>,
 }
 
 decl_module! {
 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
 		fn deposit_event<T>() = default;
 
-		/// Register an identity with the hash of the signature. Ensures that
-		/// all identities are unique, so that no duplicate identities can be
-		/// registered.
-		///
-		/// Current implementation suffers from squatter attacks. Additional
-		/// implementations could provide a mechanism for a trusted set of
-		/// authorities to delete a squatted identity OR implement storage
-		/// rent to disincentivize it.
-		pub fn register(origin, identity: Vec<u8>) -> Result {
+		pub fn register(origin, identity_type: IdentityType, identity: Vec<u8>) -> Result {
 			let _sender = ensure_signed(origin)?;
 			ensure!(!Self::frozen_accounts().iter().any(|i| i == &_sender.clone()), "Sender account is frozen");
-
-			let hash = T::Hashing::hash_of(&identity);
+			ensure!(!<UsedTypes<T>>::get(_sender.clone()).iter().any(|i| i == &identity_type), "Identity type already used");
+			let mut types = <UsedTypes<T>>::get(_sender.clone());
+			types.push(identity_type.clone());
+			<UsedTypes<T>>::insert(_sender.clone(), types);
+
+			// Hash the identity type with the identity to use as a key for the mapping
```

Comment:
I suggest just using a tuple key instead of the hash, maps support constructs like `(AccountId, Vec<u8>) => IdentityRecord`